### PR TITLE
Enhance the tweets view of TwitterStreamAgent

### PR DESCRIPTION
**Pull request by knu**
The tweets view of TwitterStreamAgent lacks some essential properties of each tweet necessary to identify the tweet and the corresponding event, so I'd like to enhance it.

- Use the `extended_tweet.full_text` property instead of `text`
- Show the time and links to the tweet and event for each tweet
- Clicking on the tweet text replaces it with the embedded tweet widget

### CODE REVIEW

File: app/views/agents/agent_views/twitter_stream_agent/_show.html.erb [hunk 1]
```diff
@@ -1,20 +1,22 @@
...
         <% group.each do |event| %>
-          <% next unless event.payload[:text].present? %>
-          <div class='tweet'>
-            <%= event.payload[:text] %> - <%= link_to event.payload[:user][:screen_name], "http://twitter.com/#{event.payload[:user][:screen_name]}" %>
-          </div>
+          <% tweet = event.payload %>
+          <% text = tweet.dig(:extended_tweet, :full_text) %>
+          <% next unless text.present? %>
+          <% screen_name = tweet.dig(:user, :screen_name) %>
+          <div class='tweet'><span class="tweet-body" data-tweet-id='<%= tweet[:id_str] %>'><span class="tweet-plain" title="Click to expand"><%= link_to "@#{screen_name}", "https://twitter.com/#{screen_name}" %>: <%= text %> - <%= link_to l(Time.zone.parse(tweet[:created_at]), format: :long), "https://twitter.com/#{screen_name}/status/#{tweet[:id_str]}" %></span></span> (<%= link_to 'event', event_path(id: event.id) %>)</div>
```
Comment (cantino) [inline app/views/agents/agent_views/twitter_stream_agent/_show.html.erb]: Do we need to escape `screen_name` in the URLs?

**Comment by knu** [comment]
@cantino Thanks for the quick check!

File: app/views/agents/agent_views/twitter_stream_agent/_show.html.erb (hunk 1 above)
Comment (knu) [reply app/views/agents/agent_views/twitter_stream_agent/_show.html.erb]: :blush:

File: app/assets/javascripts/tweets.js.coffee [hunk 2]
```diff
@@ -0,0 +1,10 @@
+$ ->
+  $('.tweet-body').each ->
+    $(this).click ->
+      twttr.widgets.createTweet(
```
Comment (dsander) [inline app/assets/javascripts/tweets.js.coffee]: Can we disable the `click` event before starting to load the widget? Sometimes it takes a good second to load and clicking on the tweet again will then load two widgets for the same tweet.

Comment (dsander) [reply app/assets/javascripts/tweets.js.coffee]: Nice thanks

**Pull request merged**